sympy/series/series.py

```diff
@@ -61,3 +63,109 @@ def series(expr, x=None, x0=0, n=6, dir="+"):
     """
     expr = sympify(expr)
     return expr.series(x, x0, n, dir)
+
+
+def inversion_series(f,x,a,n):
+    r"""
+    Computes the Taylor series of the inverse function.
+
+    The formula was proved by Lagrange (1736--1813) and generalized by the German
+    mathematician and teacher Hans Heinrich Bürmann ( --1817), both in the late 18th
+    century.
+
+    Theorem: Suppose x is defined as a function of y by an equation of the form
+
+    f(x) = y
+
+    where f is analytic at a point a and f'(a) is not equal to 0.
+    Then it is possible to invert or solve the equation for x in the form of a series:
+
+    .. math::
+        g(y) = a + \sum_{n = 1}^{oo} \frac{g_n (y - f(a))^n}{n!}
+
+    where
+
+    .. math::
+        g_n = \lim_{y \to a} \left[ \frac{d^{n-1}}{dx^{n-1}}
+        \left( \frac{x-a}{f(x)-f(a)} \right) ^n \right]
+
+    The theorem further states that this series has a non-zero radius of convergence,
+    i.e., x represents an analytic function of y in a neighbourhood of y = f(a).
+
+    Parameters
+    ==========
+
+    f : expression, the expansion of whose inverse is to be found.
+
+    x : variable, in terms of which equation is given, and in terms of which, the
+    expansion will be provided.
+
+    a : point on x-axis in the neighbourhood of which, the inverse expansion is to be found.
+
+    n : no of terms required in the expansion; default value is 3 terms.
+
+    Examples
+    ========
+
+    >>> from sympy.series.series import inversion_series
+    >>> from sympy import exp
+    >>> from sympy.abc import x
+
+    Let the equation whose inverse's expansion is to be found be y = f(x)
+    then the first argument to be supplied must be f(x).
+
+    >>> f = exp(x)
+
+    In this case, the inverse series is that of the logarithmic function around 1 = f(0):
+
+    >>> inversion_series(f,x,0,4)
+    x + (x - 1)**3/3 - (x - 1)**2/2 - 1
+
+    Functions with vanishing derivative cannot be inverted with this method
+
+    >>> f = x**2
+    >>> try:
+    ...    inversion_series(f,x,0,2)
+    ... except ValueError as ex:
+    ...     print(repr(ex))
+    ValueError('The inversion formula requires non-zero derivative.')
+
+    Note
+    ====
+
+    The implementation uses a quasi-newton method to solve an implicit equation.
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Lagrange_inversion_theorem
+    .. [2] http://mathworld.wolfram.com/LagrangeInversionTheorem.html
+    .. [3] http://www.cfm.brown.edu/people/dobrush/am33/Mathematica/lit.html
+       [4] Abramowitz, M. and Stegun, I.A., 1964. "Handbook of Mathematical Functions", Eq. 3.6.6
+    """
+    # reduce to the canonical case a=0,y=0,f'(a)=1
+    f = f.subs({x:x+a})
+    fa =f.subs({x:0})
+    dfa = f.diff(x).subs({x:0})
```

Comment:
This PR has been open for quite a while. Will I receive any complete review?

Reply:
Looking through the diff (click "files changed" tab above) there are already several comments that have not been responded to:

- The .mailmap entries are not correct.
- The comment here has not been replied to.
- The use of `inject` is questionable.

Reply:
* .mailmap is corrected now
* why the derivative and not a series - because this is how the algorithm works, it evaluates the derivative
* feel free to propose a better way than `inject` ?

Reply:
A few lines down is
```
f = f.series(x,0,n).removeO()
```
And then f is converted to Poly. At that point computing the derivative can be done much more efficiently from the Poly.

> * feel free to propose a better way than `inject`

What is the purpose of using `inject`? When I tried an example in the debugger I found that it was doing something like `ZZ.inject[3]` which means to add `3` as a generator in the polynomial ring (that should probably raise an exception because it's not meaningful). Why would you want to do that?

Reply:
If the goal is to make a Poly with a domain that includes both the coefficients of `f` and also the coefficient `1/dfa` then there are several approaches.

Using field=True ensures that the domain is a field:
```
f.as_poly(x, field=True)
```
It's clear that any domain containing the coefficients of `f` contains the coefficients `f'` so if the domain is a field then it also contains `1/dfa`.

Another possibility is
```
parallel_poly_from_expr([f, 1/dfa], x)
```
this converts both `f` and `1/dfa` to Poly with a common domain that can represent both.

You can also unify domains:
```
f.as_poly(x).unify((1/dfa).as_poly(x))
```